core/src/main/resources/runtime.js

```diff
@@ -164,12 +164,16 @@ const bytecoder = {
             Ljava$lang$ClassLoader$$getClassLoader$$: function (classRef) {
                 return null;
             },
-            Ljava$lang$Class$$forName$Ljava$lang$String$$Z$Ljava$lang$ClassLoader$: function(className, initialize, classLoader) {
-                throw 'Not supported class for reflective access';
+            Ljava$lang$Class$$forName$Ljava$lang$String$$Z$Ljava$lang$ClassLoader$: function (className, initialize, classLoader) {
+                throw new Error('Not supported class for reflective access');
             },
             Z$desiredAssertionStatus$$: function(classRef) {
                 return false;
-            }
+            },
+            Ljava$lang$String$$getName$$: function (classRef) {
+                //I am probably missing some knowledge here. This does work but it seems that @EmulateAtRuntime should do that for me?
+                return classRef.Ljava$lang$String$$getName$$();
```

Comment:
classRef.$rt.Ljava$lang$String$$getName$$() would return the classname of `Class`.
I suspect this would cause that getName() always returns `java.lang.Class`

So you are fine with the current implementation if i understand correctly?

Reply:
I think it is fine, but I haven't touched the runtime class logic for some time. So I'll merge this and we will see how far we get :-)

Reply:
So, did you add the latest changes? I'd really like to merge this PR :-)